### ISSUE TITLE
502 Bad Gateway

### ISSUE THREAD

**Issue by tylkie** [librespot-org/librespot-java#254]
**First of all**... thank you for having developed this project to such an extent... especially while Spotify still waits for enough kudos to do so :-D Great job on everything! Now... there's just one little detail I found. I understand that this problem might not be representative enough to be inspected, but I'll try anyway.

**Describe the bug**
- When trying to load any new context using Spotify's Web API, the command is received by librespot-java but nothing happens.
- The Web API responds with an error code saying "502 Bad Gateway".

**Side notes**
- All Web API commands work flawlessly on librespot... unless provided with a new context.
- All commands (even those providing a new context) using an official spotify player work on librespot-java without problems.
- Using the Web API to control an official spotify player does not render any problems either.

**To Reproduce**
Start librespot-java and make the following Web API call using a new context...
_https://developer.spotify.com/documentation/web-api/reference/player/start-a-users-playback/_

I used this wrapper...
https://github.com/JMPerez/spotify-web-api-js/tree/v1.5.1

... and this function...
_https://doxdox.org/jmperez/spotify-web-api-js#src-spotify-web-api.js-constr.prototype.play_

**Expected behavior**
When making a Web API Play call providing a new context, librespot-java loads it.

**Screenshots/Stracktraces/Logs**

Log when controlling librespot using an official spotify player...

2020-10-08 20:34:28,261 TRACE DealerClient:97 - Received request. {mid: hm://connect-state/v1/player/command, key: e9bd86bf-19a2-4e53-a587-e3c4841ea806, pid: 159267299, sender: 6b6fa2eb1fbc5e921f90726954d34e21781d424e}
2020-10-08 20:34:28,262 DEBUG Player:258 - Received command: Play
2020-10-08 20:34:28,263 DEBUG Player:311 - Loading context (play), uri: spotify:user:spotify:playlist:37i9dQZF1DX8AliSIsGeKd
2020-10-08 20:34:28,286 TRACE StateWrapper:1282 - Loaded all tracks before shuffling (entirely).
2020-10-08 20:34:28,286 TRACE StateWrapper:1290 - Shuffled context entirely!
2020-10-08 20:34:28,294 TRACE MercuryClient:138 - Send Mercury request, seq: 6, uri: hm://event-service/v1/events, method: POST
2020-10-08 20:34:28,296 TRACE MixingLine:169 - Toggle first channel: false
2020-10-08 20:34:28,296 DEBUG PlayerQueueEntry:209 - PlayerQueueEntry{013f3091194d53d3596591fdb327fef3} has been removed from output.
2020-10-08 20:34:28,297 TRACE PlayerQueue:141 - Queue has been cleared.
2020-10-08 20:34:28,297 INFO  PlayerSession:53 - Created new session. {id: aWyxDBlz72ZL9Pae4VlodA}
2020-10-08 20:34:28,298 TRACE PlayerQueueEntry:325 - PlayerQueueEntry{013f3091194d53d3596591fdb327fef3} terminated.
2020-10-08 20:34:28,298 TRACE PlayerQueueEntry:76 - Created new PlayerQueueEntry{01add2ca8d2693b2ebc118dc74e9c36e}.

Log when controlling librespot using a Web API call... (librespot just keeps preloading the chunks for the current song)

2020-10-08 20:35:53,521 TRACE DealerClient:97 - Received request. {mid: hm://connect-state/v1/player/command, key: 08d4a713-50e4-4c86-91d2-acec7744ab38, pid: 159352573, sender: webapi-453b9b9e612e4d27bdc2ea786e2562a3}
2020-10-08 20:35:53,522 DEBUG Player:258 - Received command: Play
2020-10-08 20:35:53,523 DEBUG Player:311 - Loading context (play), uri: spotify:playlist:37i9dQZF1DX8AliSIsGeKd
2020-10-08 20:35:54,473 TRACE CdnManager:262 - Chunk 16/90 completed, cached: false, stream: {fileId: BDEDC1289F41E5165681E5CD0D80E3F518CCED0B}
2020-10-08 20:36:01,206 TRACE CdnManager:262 - Chunk 17/90 completed, cached: false, stream: {fileId: BDEDC1289F41E5165681E5CD0D80E3F518CCED0B}
2020-10-08 20:36:08,014 TRACE CdnManager:262 - Chunk 18/90 completed, cached: false, stream: {fileId: BDEDC1289F41E5165681E5CD0D80E3F518CCED0B}
2020-10-08 20:36:14,571 TRACE CdnManager:262 - Chunk 19/90 completed, cached: false, stream: {fileId: BDEDC1289F41E5165681E5CD0D80E3F518CCED0B}

**Version/Commit**
librespot-java version used was 1.5.1 release without any alterations

**Comment by devgianlu** [comment]
Sorry for the late response, I'll have a look and let you know.

**Comment by devgianlu** [comment]
@tylkie I've made some changes to fix a null exception, let me know if this fixes it (it did for me). You need to compile librespot-java manually or use the latest snapshot version.

**Comment by tylkie** [comment]
Thanks! Works for me as well as far as the context change is concerned. The player does not create a new session automatically though. After a context change, one has to send a seperate play command without any context to start the new stream. Just a matter of changing some defaults I guess.

**Comment by devgianlu** [comment]
That should be fixed too.

**Comment by tylkie** [comment]
Confirmed. I guess there is no reason to rely on the official players anymore. I've been watching the entire librespot effort for a few years now. I tried every single fork and solution... librespot-java is not only the most sophisticated one, it is by far the most elegant and efficient one, too. Thanks again for your time and work! Just gonna give it a few more days to prove itself stable on my system...
